sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/Crashes.java

```diff
@@ -601,13 +609,11 @@ synchronized Device getDeviceInfo(Context context) throws DeviceInfoHelper.Devic
      *
      * @return Device information.
      */
-    private synchronized Device getLastDeviceInfo() throws JSONException {
-        String json = SharedPreferencesManager.getString(PREF_KEY_LAST_DEVICE_INFO);
-        Device lastDevice = null;
-        if (!TextUtils.isEmpty(json)) {
-            lastDevice = mLogSerializer.deserializeDevice(json);
-        }
-        return lastDevice;
+    private synchronized Device getDeviceInfoByTimestamp(Long timestamp) throws JSONException, DeviceInfoHelper.DeviceInfoException {
```

Comment:
Please use the same algorithm as session tracker, if you have a sorted set it's a 1 liner.